src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -555,7 +555,13 @@ void QgsRelationReferenceWidget::init()
 
     // Only connect after iterating, to have only one iterator on the referenced table at once
     connect( mComboBox, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, &QgsRelationReferenceWidget::comboReferenceChanged );
-    updateAttributeEditorFrame( mFeature );
+
+    if ( !mFeatureInitialized )
+    {
+      //call it for the first initialization
+      emit mComboBox->currentIndexChanged( mComboBox->currentIndex() );
```

Comment:
Is this triggering the `getFeature()` in the slot?
In this case it's executed on the main thread anyway, so doesn't make a big difference to the previous approach.

I would propose the following approach:

 * Use this approach with emitting the signal (this keeps code compact in the slot)
 * Remove the initialized condition here
 * Guard the whole `init()` function with a `mInitialized` variable (and not only the emission of this signal)

```
void init()
{
   if ( mInitialized )
     return;

  ...
  mInitialized = true;
}
```